File: Python/001-Two_Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        #创建空字典
        dict = {}
        for i in range(len(nums)):
            x = target - nums[i]
            if x in dict:
                return dict[x],i
            else:
                dict[nums[i]] = i

File: Go/001-two-sum.go
func twoSum(nums []int, target int) []int {
    var res = []int {-1, -1}
    //以<数字, 下标>形式存储
    numMap := make(map[int] int)
    
    for key, value := range nums {
        another, ok := numMap[target - value]
        if(ok) {
            res[0], res[1] = key, another
        } else {
            numMap[value] = key
        }
    }
    return res
}